plugins/partiql-plugin/src/main/kotlin/org/partiql/plugin/internal/fn/scalar/BinaryOpHelpers.kt

```diff
@@ -0,0 +1,126 @@
+package org.partiql.plugin.internal.fn.scalar
+
+import org.partiql.value.ClobValue
+import org.partiql.value.DecimalValue
+import org.partiql.value.Float32Value
+import org.partiql.value.Float64Value
+import org.partiql.value.Int16Value
+import org.partiql.value.Int32Value
+import org.partiql.value.Int64Value
+import org.partiql.value.Int8Value
+import org.partiql.value.IntValue
+import org.partiql.value.PartiQLValue
+import org.partiql.value.PartiQLValueExperimental
+import org.partiql.value.StringValue
+import org.partiql.value.SymbolValue
+import org.partiql.value.check
+import org.partiql.value.clobValue
+import org.partiql.value.decimalValue
+import org.partiql.value.float32Value
+import org.partiql.value.float64Value
+import org.partiql.value.int16Value
+import org.partiql.value.int32Value
+import org.partiql.value.int64Value
+import org.partiql.value.int8Value
+import org.partiql.value.intValue
+import org.partiql.value.stringValue
+import org.partiql.value.symbolValue
+import java.math.BigDecimal
+import java.math.BigInteger
+
+@OptIn(PartiQLValueExperimental::class)
+@JvmName("binaryOpInt8ByteToInt")
+internal inline fun binaryOpInt8(lhs: PartiQLValue, rhs: PartiQLValue, op: (Byte, Byte) -> Int): Int8Value {
+    val lhsValue = lhs.check<Int8Value>().value!!
+    val rhsValue = rhs.check<Int8Value>().value!!
+    return int8Value((op(lhsValue, rhsValue)).toByte())
+}
+
+@OptIn(PartiQLValueExperimental::class)
+@JvmName("binaryOpInt8ByteToByte")
+internal inline fun binaryOpInt8(lhs: PartiQLValue, rhs: PartiQLValue, op: (Byte, Byte) -> Byte): Int8Value {
+    val lhsValue = lhs.check<Int8Value>().value!!
+    val rhsValue = rhs.check<Int8Value>().value!!
+    return int8Value((op(lhsValue, rhsValue)))
+}
```

Comment:
As discussed offline, we won't have these helpers. See 

```
        val l = args[0].check<Int16Value>()
        val r = args[1].check<Int16Value>()
        return if (l.isNull || r.isNull) {
            boolValue(null)
        } else {
            boolValue(l.value!! > r.value!!)
        }
```

Now the question is should the null check go before or after the type check?  Seems null checks should be first.

```
1 + stringValue(null)  <-> 1 + string.null <-> 1 + null  => null
```